### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by yohannesdagnachew**
### Description

 Create a method called current_user in your ApplicationController to make current user data available to all controllers:

      > It will return the first user from the database.
      > Don't worry about proper authentication at this time.
    
Create forms to perform the following functions:
   - Create a Post on behalf of the current_user (use the method you created in your ApplicationController).
   - Create a comment on behalf of the current_user (use the method you created in your ApplicationController).

Allow Users to add likes to Posts.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,9 +2,81 @@ class PostsController < ApplicationController
...
+    puts new_comment.errors.full_messages
+    if new_comment.save
+      new_comment.update_comments_counter
+      flash[:success] = 'Success: Comment created successfully'
+      redirect_to user_posts_path(user_id: current_user.id, id: post.id)
+    else
+      flash.now[:error] = 'Error: Comment creation failed'
+      render :show, locals: { post: }
+    end
+  end
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb]: Great job @yohannesdagnachew 

- [ ] Please create a `controllers/comment_controller.rb` file to handle comment actions.
- [ ] Please there should be a pair of routes in the `config/routes.rb` file for creating comments.
- [ ] There should be a form in the `views/comments/new.html.erb` file which accepts input for creating comments.
- [ ] Please make use of the [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) in your `create_comment` method as explain in the `post_create` method below.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -2,9 +2,81 @@ class PostsController < ApplicationController
...
+
+    else
+      flash[:success] = 'Success: you disliked this post'
+      like = Like.where(post:, author_id: user.id)[0]
+      like.decreament_likes_counter
+      like.destroy
+    end
+
+    redirect_to user_post_path(user_id: current_user.id)
   end
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb]: Great job @yohannesdagnachew 

- [ ] Please create a `controllers/likes_controller.rb` file to handle comment actions.
- [ ] Please there should be a pair of routes in the `config/routes.rb` file for creating like.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -2,9 +2,81 @@ class PostsController < ApplicationController
...
+          # Success message
+          flash.now[:success] = 'Post created successfully'
+          redirect_to user_posts_path(current_user.id)
+        else
+          # Error message
+          flash.now[:error] = 'Error: Post creation failed'
+          render :new, locals: { post: new_post }
+        end
+      end
+    end
```
Comment (CollinsTatang) [inline app/controllers/posts_controller.rb:38]: Great job @yohannesdagnachew your create post method work as requested

- [ ] Please make use of the [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md) in your create post method.
### **Sample Example**
```ruby
def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
    end

   
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```
- [ ] Please focus on the `post_params` method that handles all the post attributes and makes them private increasing the security of your form inputs.

**Review by CollinsTatang — CHANGES_REQUESTED** [review]
Hi @yohannesdagnachew, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

## Required changes 
It's ok to have bugs in your code. Please, see this code review as a chance for you to learn practical coding skills and learn new tips that can improve your code quality and help you get a better job.
_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @yohannesdagnachew 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**